README.md

```diff
@@ -146,101 +159,112 @@ Following are different directives currently available.
  */
 public interface Step {
   /**
-   * Executes a wrangle step on single {@link Row} and return an array of wrangled {@link Row}.
+   * Executes a wrangle step on each {@link Row} and returns an array of wrangled {@link Row Rows}.
    *
-   * @param records List of input {@link Record} to be wrangled by this step.
-   * @return Wrangled list of {@link Record}.
-   * @throws StepException In case of any issue this exception is thrown.
+   * @param records the list of input {@link Record Records} to be wrangled by this step
+   * @return the list of wrangled {@link Record Records}
+   * @throws StepException if a step exception occurred
    */
   List<Record> execute(List<Record> records) throws StepException;
 }
 ```
 
-### Step 2/3
-Add comprehensive test case for testing the directive that has been added. 
-
-### Step 3/3
+### Part 2 of 3
+Add a comprehensive test case for testing the directive that has been added.
 
+### Part 3 of 3
 Modify the specification to parse the directive specification and create the implementation of
-Step you have created above.
+the step you have created in part 1.
 
 ## Build
-To build your plugins:
+To build your plugin:
 
     mvn clean package -DskipTests
 
-The build will create a .jar and .json file under the ``target`` directory for wrangler-transform and .jar for wrangler-service application. These files can be used to deploy your plugins and wrangler backend.
+The build will create a .jar and .json file under the `target` directory for the
+`wrangler-transform` and a .jar file for the `wrangler-service` application. These files can be used to
+deploy your plugin and the wrangler backend.
 
 
 ## Deployment
-You can deploy your plugins using the CDAP CLI:
+You can deploy your plugin using the CDAP CLI:
 
     > load artifact <target/wrangler-transform-<version>.jar> config-file <target/wrangler-transform-<version>.json>
 
-For example, if your artifact is named 'wrangler-transform-<version>':
+For example, if your artifact is named `wrangler-transform-1.0.0-SNAPSHOT`:
+
+    > load artifact target/wrangler-transform-1.0.0-SNAPSHOT.jar config-file target/wrangler-transform-1.0.0-SNAPSHOT.json
 
-    > load artifact target/wrangler-transform-<version>.jar config-file target/wrangler-transform-<version>.json
 
 ## Mailing Lists
 
 CDAP User Group and Development Discussions:
 
-- `cdap-user@googlegroups.com <https://groups.google.com/d/forum/cdap-user>`__
+* [cdap-user@googlegroups.com](https://groups.google.com/d/forum/cdap-user)
 
 The *cdap-user* mailing list is primarily for users using the product to develop
-applications or building plugins for appplications. You can expect questions from 
-users, release announcements, and any other discussions that we think will be helpful 
+applications or building plugins for appplications. You can expect questions from
+users, release announcements, and any other discussions that we think will be helpful
 to the users.
 
 ## IRC Channel
```

Comment:
Yes, or maybe replace the IRC with the Slack channel? @nitinmotgi , what do you think? Both, or  replace IRC with the Slack channel? And do this everywhere?